File: Programmers_Python/lv2_스킬트리.py
def solution(skill, skill_trees):

    cnt = 0
    for word in skill_trees:
        idx = 0
        flag = True
        for w in word:
            if w in skill:
                if skill.index(w) > idx:
                    flag = False
                    break
                idx += 1
        if flag:
            cnt+=1
    print(cnt)





if __name__ == '__main__':
    skill = "CBD"
    skill_trees = ["BACDE", "CBADF", "AECB", "BDA"]

    solution(skill, skill_trees)

    # LST..?
    # 주어진 스킬트리 순서를 지키면서 문자열이 진행되는지
    # skilll모두 안해도 됨

File: Programmers_Java/Programmers_Java/src/lv2_스킬트리.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.Set;

public class lv2_스킬트리 {
    public static int solution(String skill, String[] skill_trees) {
        Set<Character> sk = new HashSet<>();
        for (char c : skill.toCharArray()) {
            sk.add(c);
        }

        int cnt = 0;
        for (String st : skill_trees) {
            StringBuilder sb = new StringBuilder();
            for (char c : st.toCharArray()) {
                if (sk.contains(c)) sb.append(c);
            }
            if (skill.startsWith(sb.toString())) cnt++;
        }

        return cnt;
    }
    public static void main(String[] args) {
        String skill = "CBD";
        String[] skill_trees = {"BACDE", "CBADF", "AECB", "BDA"};
        solution(skill, skill_trees);
    }
}
